### PR TITLE
New module - s3_logging

### PR DESCRIPTION
**Pull request by wimnat**
New module to manage logging option of s3 buckets

### CODE REVIEW

**Comment by bcoca** [comment]
cc @ansible/aws

**Comment by bcoca** [comment]
cc @ansible/aws

**Comment by gregdek** [comment]
cc @ansible/aws

File: cloud/amazon/s3_logging.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    description:
+      - "Enable or disable logging."
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  target_bucket:
+    description:
+      - "The bucket to log to."
+    required: true
+    default: null
```
Comment (ghost) [inline cloud/amazon/s3_logging.py:44]: According to http://docs.ansible.com/developing_modules.html#module-checklist, default is not needed when required is true.

File: cloud/amazon/s3_logging.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+- name: Enable logging of s3 bucket mywebsite.com to s3 bucket mylogs
+  s3_logging:
+    name: mywebsite.com
+    target_bucket: mylogs
+    target_prefix: logs/mywebsite.com
+    state: present
+
+- name: Remove logging on an s3 bucket
+  s3_logging:
+    name: mywebsite.com
```
Comment (ghost) [inline cloud/amazon/s3_logging.py:66]: missing: target_bucket: mylogs here

**Comment by ghost** [comment]
@wimnat I did a some quick testing with this one and it works as advertised. Just a couple of documentation updates needed. The module is requiring target_bucket on absent so I think it should be added to the example.

**Comment by wimnat** [comment]
@rickmendes I've removed the target_bucket requirement and updated the doc to mention it is required for state=present.  I didn't want to have state=absent dependent on target_bucket.

**Comment by ghost** [comment]
@wimnat That works for me. It makes more sense to me to not have it required for state-absent. I give this a +1.

File: cloud/amazon/s3_logging.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  target_bucket:
+    description:
+      - "The bucket to log to. Required when state=present."
+    required: false
+    default: null
+  target_prefix:
+    description:
+      - "The prefix that should be prepended to the generated log files written to the target_bucket."
+    required: false
+    default: no
```
Comment (bcoca) [inline cloud/amazon/s3_logging.py]: i'm guessing the default here is null or is 'no' the prefix that gets added?

File: cloud/amazon/s3_logging.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+description:
+    - Manage logging facility of an s3 bucket in AWS
+version_added: "2.0"
+author: Rob White (@wimnat)
+options:
+  name:
+    description:
+      - "Name of the s3 bucket."
+    required: true
+    default: null
```
Comment (bcoca) [inline cloud/amazon/s3_logging.py]: no default needed when required is true

File: cloud/amazon/s3_logging.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+   
+    module.exit_json(changed=changed)
+    
+    
+def main():
+    
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name = dict(required=True, default=None),
```
Comment (bcoca) [inline cloud/amazon/s3_logging.py]: default is never used here as you require the input

File: cloud/amazon/s3_logging.py (hunk 3 above)
Comment (wimnat) [inline cloud/amazon/s3_logging.py]: Should be an empty string rather than null.

**Pull request merged**